spec/order_spec.rb

```diff
@@ -0,0 +1,64 @@
+require 'order'
+
+describe Order do
```

Comment:
There's quite a lot to take in here, in one describe block. Perhaps you could break up into seperate describe blocks for each method being tested? And if necessary add context blocks.